src/middlewares/Mock.ts

```diff
@@ -0,0 +1,73 @@
+import {IRequest, IResponse} from "../Client";
+import {IMiddleware} from "../Stack";
+import {FetchResponse} from "./Fetch";
+
+export interface IMockHandler<IN, OUT> {
+    match: (options: IN) => boolean|undefined;
+    factory: (options: IN) => OUT;
+    delay?: number;
+}
+
+declare const Response: FetchResponse;
+
+export class Mock implements IMiddleware<IRequest, Promise<IResponse<any>>> {
+
+    public defaultDelay: number = 5;
+
+    public constructor(protected handlers: IMockHandler<IRequest, Promise<IResponse<any>>|IResponse<any>>[] = []) { }
+
+    public static jsonResponse<T>(data: T, response?: IResponse<any>): IResponse<T> {
+
+        // Encode data
+        const stream: string|undefined = (data === undefined || data === null) ? undefined : JSON.stringify(data);
+
+        // Create Data
+        const init = Object.assign(
+            {},
+            {
+                status: data === undefined || data === null ? 204 : 200,
+                headers: {
```

Comment:
Please research: I think the Content-Type header is omitted for 204 (because No-Content implies no content type)